src/controls/events/move-input.js

```diff
@@ -16,65 +17,79 @@ export default class MoveInput {
     this.callback = callback;
     this.pressed = false;
 
-    const events = MOUSE_EVENTS.concat(options.events || []);
-    this.options = Object.assign({enable: true}, options, {events});
+    this.options = Object.assign({enable: true}, options);
+    this.enableMoveEvent = this.options.enable;
+    this.enableLeaveEvent = this.options.enable;
+
+    this.events = MOUSE_EVENTS.concat(options.events || []);
 
     this.handleEvent = this.handleEvent.bind(this);
-    this.options.events.forEach(event => element.addEventListener(event, this.handleEvent));
+    this.events.forEach(event => element.addEventListener(event, this.handleEvent));
   }
 
   destroy() {
-    this.options.events.forEach(event => this.element.removeEventListener(event, this.handleEvent));
-  }
-
-  set(options) {
-    Object.assign(this.options, options);
+    this.events.forEach(event => this.element.removeEventListener(event, this.handleEvent));
   }
 
   /**
    * Enable this input (begin processing events)
    * if the specified event type is among those handled by this input.
    */
   toggleIfEventSupported(eventType, enabled) {
-    if (EVENT_TYPE === eventType) {
-      this.options.enable = enabled;
+    if (eventType === MOVE_EVENT_TYPE) {
+      this.enableMoveEvent = enabled;
```

Comment:
This makes more sense, I couldn't see how the old system could be made to fit...